src/Common/src/System/Net/Security/SslClientAuthenticationOptionsExtensions.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Net.Security
+{
+    internal static class SslClientAuthenticationOptionsExtensions
+    {
+        public static SslClientAuthenticationOptions ShallowClone(this SslClientAuthenticationOptions options) =>
+            new SslClientAuthenticationOptions()
+            {
+                AllowRenegotiation = options.AllowRenegotiation,
+                ApplicationProtocols = options.ApplicationProtocols,
+                CertificateRevocationCheckMode = options.CertificateRevocationCheckMode,
+                ClientCertificates = options.ClientCertificates,
+                EnabledSslProtocols = options.EnabledSslProtocols,
+                EncryptionPolicy = options.EncryptionPolicy,
+                LocalCertificateSelectionCallback = options.LocalCertificateSelectionCallback,
+                RemoteCertificateValidationCallback = options.RemoteCertificateValidationCallback,
+                TargetHost = options.TargetHost
```

Comment:
TargetHost should not be externally configured. Ignore it in the clone?

HttpClient is designed to connect concurrently to multiple hosts. This value should be taken from the Host header or Url of the HttpRequestMessage creating the connection.

Reply:
For these where the implementation may want to specify its own values, my thinking was we allow the developer to specify, and if they do, it overrides whatever the implementation wanted to do, and if the developer explicitly specifies something / specifies something non-sensical, that's their problem.  That's why here, for example:
https://github.com/dotnet/corefx/pull/26870/files#diff-85488c8014383a118e955b5f28c7ac15R114
it does:
```C#
if (sslOptions.TargetHost == null) sslOptions.TargetHost = host;
```
such that if the dev hasn't explicitly changed TargetHost, we'll use the host from the host header / url, but if they have, then we'll use what they provided.